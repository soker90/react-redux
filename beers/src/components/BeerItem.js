import React, { Component } from 'react'

class BeerItem extends Component {
  render() {
    const {beer, onRemove} = this.props
    const btnText = this.props.loading ? 'Loading...' : 'Remove Beer'

    return (
      <div>
        <p>
          <b>{beer.name}</b> ({beer.graduation}%) <i>{beer.id}</i>
        </p>
        <div>
          <button onClick={onRemove} disabled={this.props.loading}>{btnText}</button>
        </div>
      </div>
    )
  }
}


export default BeerItem
