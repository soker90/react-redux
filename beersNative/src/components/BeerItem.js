import React, { Component } from 'react'
import { Text, TouchableOpacity, View } from 'react-native'

class BeerItem extends Component {
  render() {
    const {beer, onRemove} = this.props
    const btnText = this.props.loading ? 'Loading...' : 'Remove Beer'


    return (
      <View>
        <Text>
          {beer.name} ({beer.graduation}%) - {beer.price}â‚¬ - {beer.id}
        </Text>
        <View>
          <TouchableOpacity onPress={onRemove} disabled={this.props.loading}><Text>{btnText}</Text></TouchableOpacity>
        </View>
      </View>
    )
  }
}



export default BeerItem
