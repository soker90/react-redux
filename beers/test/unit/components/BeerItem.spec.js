// BeerItem.spec.js
import React from 'react'
import renderer from 'react-test-renderer'

import BeerItem from '../../../src/components/BeerItem'
import mockData from '../mockData'

describe('BeerItem component', () => {
  test('render BeerItem component without crashing', () => {
    const beer = mockData.beer
    const loading = false
    const onRemove = jest.fn()

    const tree = renderer.create(
      <BeerItem beer={beer} onRemove={onRemove} loading={loading}/>
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })

  test('render BeerItem component with loading true', () => {
    const beer = mockData.beer
    const loading = true
    const onRemove = jest.fn()

    const tree = renderer.create(
      <BeerItem beer={beer} onRemove={onRemove} loading={loading}/>
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })

})
