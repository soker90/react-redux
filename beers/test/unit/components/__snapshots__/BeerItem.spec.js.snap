exports[`BeerItem component render BeerItem component with loading true 1`] = `
<div>
  <p>
    <b>
      Amstel
    </b>
     (
    5
    %) 
    <i>
      beer_01
    </i>
  </p>
  <div>
    <button
      disabled={true}
      onClick={[Function]}>
      Loading...
    </button>
  </div>
</div>
`;

exports[`BeerItem component render BeerItem component without crashing 1`] = `
<div>
  <p>
    <b>
      Amstel
    </b>
     (
    5
    %) 
    <i>
      beer_01
    </i>
  </p>
  <div>
    <button
      disabled={false}
      onClick={[Function]}>
      Remove Beer
    </button>
  </div>
</div>
`;
